internal/services/network/network_interface_data_source.go

```diff
@@ -264,5 +272,9 @@ func dataSourceNetworkInterfaceRead(d *pluginsdk.ResourceData, meta interface{})
 		d.Set("enable_accelerated_networking", props.EnableAcceleratedNetworking)
 	}
 
-	return tags.FlattenAndSet(d, resp.Tags)
```

Comment:
Doing that will let us keep this line the same without doing what you've done below